pkg/builders/s2i/builder.go

```diff
@@ -139,7 +139,7 @@ func (b *Builder) Build(ctx context.Context, f fn.Function, platforms []fn.Platf
 	// Build Config
 	cfg := &api.Config{}
 	cfg.Quiet = !b.verbose
-	cfg.Tag = f.Build.Image
+	cfg.Tag = f.Image
```

Comment:
Is it changing the output of si2 build right?

Reply:
The build config's `cfg.Tag` should be set to `f.Image` if it is defined (this indicates a request to build exactly this image using, for example `--image=` flag via the CLI.  If that is not defined, it should use `f.Build.Image`  (the last built image) if it exists, which is essentially rebuilding.  It seems like there might be a little missing logic here to choose one or the other?  Am I seeing this correctly @gauron99 ?

Reply:
Thanks for the explanation! 
So I'd say it's a change unrelated to image version update then. Probably easier to address in an isolated issue/PR? Unless it was trying to solve something else during testing with NodeJS 18 image @matejvasek?

Reply:
client builder should have set the `f.Build.Image` to whatever the final resolved image is supposed to be (using forced image via `f.Image` or resolved via registry name etc.), so Im pretty sure the `f.Build.Image` is correct here